assets/src/scripts/monitoring-location/components/hydrograph/hydrograph-utils.js

```diff
@@ -59,3 +63,41 @@ export const parsePeriodCode = function(periodCode) {
         };
     }
 };
+
+/*
+ * Converts a celsius time series to fahrenheit and store in collection.
+ * @param {Object} time series collection
+ * @returns {Object} time series collection with fahrenheit version added
+ */
+export const convertTemperatureSeriesAndAddToCollection = function(collection) {
+    Object.entries(collection.timeSeries).forEach((currentInLoopSeries) => {
```

Comment:
This confused the heck out of me until I ran the code. I think it would be clearer if you code this line as follows:
```suggestion
    Object.entries(collection.timeSeries).forEach((tsRequestKey, timeSeries) => {
```
Then the next line could be:
```suggestion
    const variableCode = timeSeries.variable;
```
The rest of the function would be clearer too, particular where you are creating the calculated timeSeries tsRequestKey.

Reply:
I completely refactored this, so (hopefully) it is easier to see what is happening and the original 'collection' is not changed and only the converted objects are added to it.